tests/util/server/src/lsp.rs

```diff
@@ -534,6 +549,9 @@ impl LspClientBuilder {
       .arg("lsp")
       .stdin(Stdio::piped())
       .stdout(Stdio::piped());
+    for (key, value) in &self.env_specified {
+      command.env(key, value);
+    }
```

Comment:
Oh, this probably should be inheriting the environment from the `TestContext` similar to how `new_command()` works. That said, this is good to have too.